api/forms.py

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+from django import forms
+from django.forms.widgets import (TextInput, Textarea, RadioSelect, CheckboxSelectMultiple,
+                                 MultiWidget)
+from crispy_forms.helper import FormHelper
+from crispy_forms.utils import render_crispy_form
+
+
+class HopperForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        model = kwargs.pop('model', None)
+        elements = model.convert_to_dict(model.elements)
+        super(HopperForm, self).__init__(*args, **kwargs)
+        self.type_widget_mapping = {
+            'input': TextInput,
+            'fieldset': MultiWidget,
+            'textarea': Textarea,
+            'radio': RadioSelect,
+            'checkbox': CheckboxSelectMultiple,
+        }
```

Comment:
Until now we don't have an exact API definition for all widget types, so I only added that one I tested.